scripts/scil_project_map_to_streamlines.py

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+"""
+Projects metrics extracted from a map onto the endpoints of streamlines.
+
+The default options will take data from a nifti image (3D or ND) and
+project it onto the endpoints of streamlines.
+"""
+
+import argparse
+import logging
+from copy import deepcopy
+
+from numpy import asarray, reshape, repeat
+
+from scilpy.io.image import load_img
+from scilpy.io.streamlines import load_tractogram_with_reference
+from scilpy.io.utils import (add_overwrite_arg,
+                             assert_inputs_exist,
+                             assert_outputs_exist,
+                             add_reference_arg,
+                             add_verbose_arg)
+
+from scilpy.image.volume_space_management import DataVolume
+
+from dipy.io.streamline import save_tractogram, StatefulTractogram
+
+
+def project_metric_to_streamlines(sft, metric, endpoints_only=False):
+    """
+    Projects a metric onto the points of streamlines.
+
+    Parameters
+    ----------
+    sft: StatefulTractogram
+        Input tractogram.
+    metric: DataVolume
+        Input metric.
+
+    Optional:
+    ---------
+    endpoints_only: bool
+        If True, will only project the metric onto the endpoints of the
+        streamlines (all values along streamlines set to zero). If False,
+        will project the metric onto all points of the streamlines.
+
+    Returns
+    -------
+    streamline_data:
+        metric projected to each point of the streamlines.
+    """
+    if len(metric.data.shape) == 4:
+        dimension = metric.data.shape[3]
+    else:
+        dimension = 1
+
+    streamline_data = []
+    if endpoints_only:
+        for s in sft.streamlines:
+            p1_data = metric.get_value_at_coordinate(
+                s[0][0], s[0][1], s[0][2], space=sft.space, origin=sft.origin)
+            p2_data = metric.get_value_at_coordinate(
+                s[-1][0], s[-1][1], s[-1][2], space=sft.space, origin=sft.origin)
+            thisstreamline_data = []
+            for p in s:
+                thisstreamline_data.append(
+                    asarray(repeat(0, p1_data.shape[0])))
```

Comment:
p1_data.shape[0] causes an error if the metric image is not multivalued. A fix would be to add a condition: 
```
if dimension == 1: 
       thisstreamline_data.append(0)
```